tower/src/util/combinators/map_err.rs

```diff
@@ -0,0 +1,64 @@
+use futures_util::{future::MapErr as MapErrFut, TryFutureExt};
+use std::marker::PhantomData;
+use std::task::{Context, Poll};
+use tower_layer::Layer;
+use tower_service::Service;
+
+#[derive(Debug)]
+pub struct MapErr<S, F> {
```

Comment:
It would be nice to have a brief doc comment on this struct --- it could just be 
```suggestion
/// Service returned by the [`map_err`] combinator.
///
/// [`map_err`]: ../../trait.ServiceExt.html#method.map_err
pub struct MapErr<S, F> {
```